code/game/objects/structures/xeno.dm

```diff
@@ -709,6 +718,13 @@ TUNNEL
 	else
 		to_chat(M, "<span class='warning'>Our crawling was interrupted!</span>")
 
+/obj/structure/tunnel/proc/hud_set_xeno_tunnel() //called by the Dig Tunnel ability after creation; make this handy overlay exclusively visible to xenos of the same hive.
+	var/image/tunneloverlay = image('icons/mob/hud.dmi', src, "hudtraitor")
+
+	for(var/mob/living/carbon/xenomorph/X in creator.hive.get_all_xenos()) //Set hive specific image visibility
+		X.client.images += tunneloverlay
+
```

Comment:
I dont think this works for xenos joining after the tunnel is made
This not using HUD code properly is now my only qualm currently though

Reply:
You might be right.

As mentioned before though, and on discord, if we could find a way to make the HUD code work, I would do it in an instant.

Reply:
Well fuck it; I'm going to use the code that *should* work instead.